src/components/InputField.tsx

```diff
@@ -1,123 +1,68 @@
 
 type Props = {
+  type?: Types
   title: string
-  value: string
-  setValue: React.Dispatch<React.SetStateAction<string>>
-  type?: 'text' | 'name' | 'email'
+  errors: { [x: string]: any }
+  register: UseFormRegister<FieldValues>
+  pattern?: RegExp
   required?: boolean
 }
 
-const warningLabels: Record<string, string> = {
-  // warning text that appears on tooltip
+const defaultProps = {
+  type: 'text',
+  pattern: defaultRegex,
+  required: false,
+}
+
+const warningLabelTexts: Record<Types, string> = {
   text: missingField,
   name: invalidName,
   email: invalidEmail,
 }
-
-const checkPatterns: Record<string, string> = {
-  // regex used to check validity of custom input types
-  default: defaultRegex,
-  name: nameRegex,
-}
-
-const customBrowserMessages: Record<string, string> = {
-  // message shown by browser onInvalid
-  default: defaultBrowserMessage,
-  name: nameBrowserMessage,
-}
-
 function InputField(props: Props) {
   const theme = useTheme()
-  const { title, value, setValue, required } = props
 
-  const { common, danger } = { ...theme.palette }
-  const { white, gray } = { ...common }
-  const { input, h2 } = { ...theme.typography.fontSize }
-
-  const [labelId, setLabelId] = useState('')
-  const [labelElement, setLabelElement] = useState<HTMLElement | null>(null) // html element of tooltip used to make tooltip visible/invisible
-
-  const warningLabel = warningLabels[props.type ?? 'text']
-  const checkPattern = checkPatterns[props.type ?? 'default'] ?? checkPatterns.default
-  const customInvalidMsg = customBrowserMessages[props.type ?? 'default'] ?? customBrowserMessages.default
-
-  useEffect(() => {
-    setLabelId(uniqueId.uniqueId('input-label-'))
-  }, [])
-
-  useEffect(() => {
-    setLabelElement(required ? document.getElementById(labelId) : null)
-  }, [labelId])
+  const { title, errors, register, required } = props
+  const type = props.type ?? 'text'
+  const pattern = props.pattern ?? defaultRegex
 
-  const onInput = (e: any, callback = false) => {
-    // supposed to be (e: React.FormEvent<HTMLInputElement>) but TS will highlight (e.target.setCustomValidity) as invalid, that's why (any)
-    if (required && labelElement) {
-      if (!callback) e.target.setCustomValidity('')
-      labelElement.style.opacity = INVISIBLE // hide tooltip
-    }
-  }
+  const {
+    common: { white },
+    primary,
+  } = { ...theme.palette }
+  const { input, h2 } = { ...theme.typography.fontSize }
 
-  const onInvalid = async (e: any) => {
-    if (required && labelElement) {
-      labelElement.style.opacity = VISIBLE // show tooltip
-      e.target.setCustomValidity(customInvalidMsg) // overrides custom message displayed by browser onInvalid
-      setTimeout(onInput, 2000, e, true) // hide tooltip after 2s
-    }
-  }
+  const warningLabelText = warningLabelTexts[`${type}`]
+  const inputInvalid = title in errors
 
+  /* eslint-disable react/jsx-props-no-spreading */
+  /* eslint-disable object-shorthand */
```

Comment:
@hui444 do we want to add these into eslintrc?

Reply:
@marcus-ong-qy yes pls, help update!